compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp

```diff
@@ -53,14 +54,26 @@ namespace mlir::iree_compiler::IREE::HAL {
 static constexpr char kQueryFunctionName[] =
     "iree_hal_executable_library_query";
 
-static void dumpBitcodeToPath(StringRef path, StringRef baseName,
-                              StringRef suffix, StringRef extension,
-                              llvm::Module &module) {
-  llvm::SmallVector<char, 0> data;
-  llvm::raw_svector_ostream ostream(data);
-  llvm::WriteBitcodeToFile(module, ostream);
-  dumpDataToPath(path, baseName, suffix, extension,
-                 StringRef(data.data(), data.size()));
+static void dumpLLVMModuleToPath(StringRef path, StringRef baseName,
+                                 StringRef suffix, StringRef extPrefix,
+                                 llvm::Module &module) {
```

Comment:
If this is duplicated between multiple files, we could move it to https://github.com/iree-org/iree/tree/main/compiler/src/iree/compiler/Dialect/HAL/Utils. @benvanik WDYT?

Reply:
I've also noticed that the smiliar code snippet has been used on various platforms. We could refactor it in the following pull request if there are no problems moving the similar functions to the path @ScottTodd provided.